pinot-query-planner/src/main/java/org/apache/pinot/query/planner/physical/v2/PRelToPlanNodeConverter.java

```diff
@@ -117,7 +118,8 @@ public static ExchangeNode convertPhysicalExchange(PhysicalExchange node) {
     return new ExchangeNode(DEFAULT_STAGE_ID, toDataSchema(node.getRowType()),
         new ArrayList<>(), node.getRelExchangeType(), RelDistribution.Type.ANY, node.getDistributionKeys(),
         false, node.getRelCollation().getFieldCollations(), false,
-        !node.getRelCollation().getKeys().isEmpty(), Set.of() /* table names */, node.getExchangeStrategy());
+        !node.getRelCollation().getKeys().isEmpty(), Set.of() /* table names */, node.getExchangeStrategy(),
+        KeySelector.DEFAULT_HASH_ALGORITHM);
```

Comment:
We need to pass the hash value through for the v2 optimizer too. I can raise a separate patch for this though since this is already a big PR